src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -186,6 +187,31 @@ public FileModel lookupPath(String datasetId, String path, int depth) {
     return fileModelFromFSItem(fsItem);
   }
 
+  public Optional<FileModel> lookupOptionalPath(String datasetId, String path, int depth) {
+    Dataset dataset = datasetService.retrieveAvailable(UUID.fromString(datasetId));
+    CloudPlatformWrapper cloudPlatformWrapper =
+        CloudPlatformWrapper.of(dataset.getDatasetSummary().getStorageCloudPlatform());
+    final Optional<FSItem> file;
+    if (cloudPlatformWrapper.isGcp()) {
+      try {
+        file = fileDao.lookupOptionalPath(dataset, path, depth);
+      } catch (InterruptedException ex) {
+        throw new FileSystemExecutionException(
+            "Unexpected interruption during file system processing", ex);
+      }
+    } else {
+      BillingProfileModel billingProfileModel =
+          profileService.getProfileByIdNoCheck(dataset.getDefaultProfileId());
+      AzureStorageAccountResource storageAccountResource =
+          resourceService.getStorageAccount(dataset, billingProfileModel);
+      AzureStorageAuthInfo storageAuthInfo =
+          AzureStorageAuthInfo.azureStorageAuthInfoBuilder(
+              billingProfileModel, storageAccountResource);
+      file = tableDao.lookupOptionalPath(UUID.fromString(datasetId), path, storageAuthInfo, depth);
```

Comment:
I think we already have the `uuid` in `dataset`
```suggestion
      file = tableDao.lookupOptionalPath(dataset.getId(), path, storageAuthInfo, depth);
```